feat(core): drop support for older TypeScript versions

crisbeto opened the pull request:
Drops support for versions of TypeScript older than 5.2

BREAKING CHANGE:
Versions of TypeScript older than 5.2 are no longer supported.

crisbeto committed:
feat(core): drop support for older TypeScript versions

Drops support for versions of TypeScript older than 5.2

BREAKING CHANGE:
Versions of TypeScript older than 5.2 are no longer supported.

crisbeto commented on aio/content/examples/upgrade-phonecat-1-typescript/tsconfig.ajs.json:
```diff
@@ -8,6 +8,6 @@
     "noImplicitAny": false,
     "skipLibCheck": true,
-    "suppressImplicitAnyIndexErrors": true
+    "ignoreDeprecations": "5.0"
```
Context: TS was throwing an error because the previous option was removed. This was the suggested replacement.

JeanMeche commented:
FYI this will break demos that rely on webcontainers as it still uses node 16.

crisbeto commented:
Hmm, good point. I can reproduce it here https://stackblitz.com/edit/57cyrd?file=.stackblitzrc.

JeanMeche commented:
Btw I reached out to the stackblitz team, they expect to support node 18 in late september.

crisbeto commented:
I'll revert the update to AIO for now and disable the TS version check instead.

crisbeto committed:
build: disable TS version checks on AIO

Disables the checks for the version of TS in AIO to unblock the TS deprecation.

jessicajaniuk reviewed (approved):
LGTM

pkozlowski-opensource commented:
This PR was merged into the repository by commit e1b27b01a5e0b5a4e4fb2954dbbd9e1fac32f4f6.
